src/types/flow.ts

```diff
@@ -19,10 +19,11 @@ export type FlowGraph = {
   [key: string]: Node;
 };
 
-export interface IndexedNode extends Node {
+export type IndexedNode = Node & {
   id: string;
+  type: ComponentType;
   parentId: string | null; // null if it is the first node
-}
+};
```

Comment:
This is a `type` rather than an `interface` because it's a known well-defined structure and shouldn't have any additional undefined fields.

`IndexedNode`s should always have a defined type.